packages/@aws-cdk/core/lib/deployment-configuration.ts

```diff
@@ -0,0 +1,635 @@
+import * as asset_schema from '@aws-cdk/cdk-assets-schema';
+import * as cxschema from '@aws-cdk/cloud-assembly-schema';
+import * as cxapi from '@aws-cdk/cx-api';
+import * as crypto from 'crypto';
+import * as fs from 'fs';
+import * as path from 'path';
+import { DockerImageAssetLocation, DockerImageAssetSource, FileAssetLocation, FileAssetPackaging, FileAssetSource } from './assets';
+import { Fn } from './cfn-fn';
+import { Construct, ConstructNode, IConstruct, ISynthesisSession } from './construct-compat';
+import { FileAssetParameters } from './private/asset-parameters';
+import { Stack } from './stack';
+import { Token } from './token';
+
+/**
+ * The well-known name for the docker image asset ECR repository. All docker
+ * image assets will be pushed into this repository with an image tag based on
+ * the source hash.
+ */
+const ASSETS_ECR_REPOSITORY_NAME = 'aws-cdk/assets';
+
+/**
+ * This allows users to work around the fact that the ECR repository is
+ * (currently) not configurable by setting this context key to their desired
+ * repository name. The CLI will auto-create this ECR repository if it's not
+ * already created.
+ */
+const ASSETS_ECR_REPOSITORY_NAME_OVERRIDE_CONTEXT_KEY = 'assets-ecr-repository-name';
+
+/**
+ * Encodes information how a certain Stack should be deployed
+ */
+export interface IDeploymentConfiguration {
```

Comment:
To me, "configuration" implies a set of values, not a bunch of operations. Maybe `IDeploymentProvider` or `IEnvironmentDeployment` or `IEnvironmentProvider`.